python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_health.py

```diff
@@ -35,11 +35,7 @@ async def get_asset_check_status_and_metadata(
         GrapheneAssetHealthStatus,
     )
 
-    asset_check_health_state = None
-    if graphene_info.context.instance.streamline_read_asset_health_supported():
-        asset_check_health_state = (
-            graphene_info.context.instance.get_asset_check_health_state_for_asset(asset_key)
-        )
+    asset_check_health_state = await AssetCheckHealthState.gen(graphene_info.context, asset_key)
```

Comment:
As an example of what I was talking about in the description about the non-streamline path, the block of code under `if asset_check_health_state is None` allows us to generate the health data if the asset isn't stored in streamline (or if streamline is disabled for the org). It feels a bit odd to expect all callers of `AssetCheckHealthState.gen` to handle the non-streamline case separately, but there's a lot of extra information I need to generate the state without streamline (like the list of asset check keys) that would need to be passed in to `gen`. So for now, just leaving it split like this. I can take another pass at cleaning it up in a stacked PR

Reply:
maybe to make it feel a bit nicer I could wrap `AssetCheckHealthState.gen` in `AssetCheckHealthState.gen_from_streamline` or something, so that it's more clear that it only works for streamline data